js/components/portal-dashboard/multiple-choice-choices.tsx

```diff
@@ -0,0 +1,125 @@
+import React from "react";
+import { Map, List } from "immutable";
+import Choice from "./choice";
+
+interface IProps {
+  answers?: Map<string, any>;
+  choices?: List<Map<string, any>>;
+  inQuestionDetailsPanel?: boolean;
+  question: Map<string, any>;
+  showStats?: boolean;
+  studentAnswer?: Map<any, any>;
+  students?: Map<any, any>;
+}
+
+function buildClassChoiceData(
+  choices: List<Map<string, any>>,
+  answers: Map<string, any>,
+  students: Map<any, any>,
+  inQuestionDetailsPanel?: boolean
+) {
+  let answerCount = 0;
+  let choiceData = choices.map((choice: any) => {
+    if (!choice) return Map();
+    const choiceId = choice.get("id");
+    const count = answers.reduce((acc: number, answer: any) => {
```

Comment:
There is just 1 answer per student so the `const noResponseCount = students.size - answerCount;` expression could never be negative right?  Just wanted to check we don't get potential negative percentages.

Reply:
I'm pretty sure there's just one answer, but just in case, I've added a guard against negative values. Thanks.